<a href="https://colab.research.google.com/github/lc21140665-lgtm/Analisis-y-visualizacion-de-datos/blob/main/Practica_01_Pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
df = pd.read_excel("Ventas_2026.xlsx")

In [3]:
df.head()

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0
1,2,2026-04-16 00:00:00,Poniente,Aguascalientes,Aguascalientes,Luis Torres,Cliente 124,Laptop,Laptop Dell,2,1231.08,914.96,Tarjeta,10,16.0
2,3,2026-05-21 00:00:00,Poniente,Celaya,Guanajuato,Ana López,Cliente 158,Monitor,Monitor Samsung,2,15290.91,9106.87,Tarjeta,10,16.0
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop HP,3,10756.13,6950.42,Transferencia,20,16.0
4,5,2026-05-19 00:00:00,Sur,Aguascalientes,Aguascalientes,Ana López,Cliente 33,Laptop,Laptop Dell,6,18571.09,12060.69,Transferencia,15,16.0


In [4]:
df.shape

(1010, 15)

In [5]:
df.columns

Index(['ID Venta', 'Fecha', 'Sucursal', 'Ciudad', 'Estado', 'Vendedor',
       'Cliente', 'Categoría', 'Producto', 'Cantidad', 'Precio Unitario',
       'Costo Unitario', 'Forma de Pago', 'Descuento (%)', 'IVA (%)'],
      dtype='object')

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1010 entries, 0 to 1009
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   ID Venta         1010 non-null   int64  
 1   Fecha            1010 non-null   object 
 2   Sucursal         1010 non-null   object 
 3   Ciudad           1010 non-null   object 
 4   Estado           1010 non-null   object 
 5   Vendedor         1010 non-null   object 
 6   Cliente          1010 non-null   object 
 7   Categoría        1010 non-null   object 
 8   Producto         1010 non-null   object 
 9   Cantidad         1010 non-null   int64  
 10  Precio Unitario  1010 non-null   float64
 11  Costo Unitario   1010 non-null   float64
 12  Forma de Pago    1010 non-null   object 
 13  Descuento (%)    1010 non-null   int64  
 14  IVA (%)          1004 non-null   float64
dtypes: float64(3), int64(3), object(9)
memory usage: 118.5+ KB


Ejercicio de comprobación
Responde las siguientes preguntas:
**1.	¿Cuántos registros tiene el archivo? **
Tiene 1,010 registros
**2.	¿Cuántas columnas contiene?**
Contiene 15 columnas
**3.	¿Cuál es el nombre de la primera columna?**
ID_Venta
**4.	¿Qué tipo de dato tiene la columna Cantidad?**
int64, es decir, un número entero
**5.	¿Existen columnas con valores nulos? ¿Cuáles? **
Sí. La única columna con valores nulos es IVA (%), con 6 valores vacíos.
**6.	¿Qué comando utilizaste para visualizar los primeros registros? **
df.head()
**7.	¿Qué comando permite conocer el tamaño del DataFrame?**
df.shape


PRACTICA 2

In [7]:
df.head()

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0
1,2,2026-04-16 00:00:00,Poniente,Aguascalientes,Aguascalientes,Luis Torres,Cliente 124,Laptop,Laptop Dell,2,1231.08,914.96,Tarjeta,10,16.0
2,3,2026-05-21 00:00:00,Poniente,Celaya,Guanajuato,Ana López,Cliente 158,Monitor,Monitor Samsung,2,15290.91,9106.87,Tarjeta,10,16.0
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop HP,3,10756.13,6950.42,Transferencia,20,16.0
4,5,2026-05-19 00:00:00,Sur,Aguascalientes,Aguascalientes,Ana López,Cliente 33,Laptop,Laptop Dell,6,18571.09,12060.69,Transferencia,15,16.0


In [8]:
df.head(10)

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0
1,2,2026-04-16 00:00:00,Poniente,Aguascalientes,Aguascalientes,Luis Torres,Cliente 124,Laptop,Laptop Dell,2,1231.08,914.96,Tarjeta,10,16.0
2,3,2026-05-21 00:00:00,Poniente,Celaya,Guanajuato,Ana López,Cliente 158,Monitor,Monitor Samsung,2,15290.91,9106.87,Tarjeta,10,16.0
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop HP,3,10756.13,6950.42,Transferencia,20,16.0
4,5,2026-05-19 00:00:00,Sur,Aguascalientes,Aguascalientes,Ana López,Cliente 33,Laptop,Laptop Dell,6,18571.09,12060.69,Transferencia,15,16.0
5,6,2026-03-02 00:00:00,Oriente,Aguascalientes,Aguascalientes,carlos ruiz,Cliente 157,Monitor,Monitor Samsung,4,22944.36,15344.97,Tarjeta,15,16.0
6,7,2026-02-02 00:00:00,Oriente,Aguascalientes,Aguascalientes,Juan Pérez,Cliente 189,Monitor,Monitor LG,10,206.32,173.02,Tarjeta,10,16.0
7,8,2026-02-07 00:00:00,Sur,Querétaro,Querétaro,Luis Torres,Cliente 98,Monitor,Monitor LG,7,14680.86,12243.79,Tarjeta,20,16.0
8,9,2026-03-28 00:00:00,Sur,San Luis Potosí,San Luis Potosí,Ana López,Cliente 69,Laptop,Laptop HP,9,18736.06,15318.53,Efectivo,0,16.0
9,10,2026-01-17 00:00:00,Poniente,Celaya,Guanajuato,Sofía Díaz,Cliente 27,Monitor,Monitor Samsung,9,15887.35,11778.45,Tarjeta,15,16.0


In [9]:
df.tail()

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
1005,898,2026-05-26 00:00:00,Centro,León,Guanajuato,sofía díaz,Cliente 111,Accesorios,Audífonos,8,1076.11,663.89,Efectivo,0,16.0
1006,882,2026-03-16 00:00:00,Centro,Celaya,Guanajuato,Juan Pérez,Cliente 124,Laptop,Laptop Lenovo,7,10036.48,7883.42,Efectivo,0,16.0
1007,499,2026-01-01 00:00:00,Centro,Aguascalientes,Aguascalientes,María Gómez,Cliente 89,Laptop,Laptop Dell,9,18873.15,14705.85,Transferencia,0,16.0
1008,135,2026-03-04 00:00:00,Oriente,Querétaro,Querétaro,Sofía Díaz,Cliente 36,Laptop,Laptop Dell,-1,4668.93,3620.85,Tarjeta,5,16.0
1009,94,2026-03-06 00:00:00,Sur,Aguascalientes,Aguascalientes,Juan Pérez,Cliente 129,Accesorios,Webcam,8,7441.66,4620.54,Transferencia,15,16.0


In [10]:
df.shape

(1010, 15)

In [11]:
df.columns

Index(['ID Venta', 'Fecha', 'Sucursal', 'Ciudad', 'Estado', 'Vendedor',
       'Cliente', 'Categoría', 'Producto', 'Cantidad', 'Precio Unitario',
       'Costo Unitario', 'Forma de Pago', 'Descuento (%)', 'IVA (%)'],
      dtype='object')

In [12]:
df.dtypes

,0
ID Venta,int64
Fecha,object
Sucursal,object
Ciudad,object
Estado,object
Vendedor,object
Cliente,object
Categoría,object
Producto,object
Cantidad,int64


In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1010 entries, 0 to 1009
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   ID Venta         1010 non-null   int64  
 1   Fecha            1010 non-null   object 
 2   Sucursal         1010 non-null   object 
 3   Ciudad           1010 non-null   object 
 4   Estado           1010 non-null   object 
 5   Vendedor         1010 non-null   object 
 6   Cliente          1010 non-null   object 
 7   Categoría        1010 non-null   object 
 8   Producto         1010 non-null   object 
 9   Cantidad         1010 non-null   int64  
 10  Precio Unitario  1010 non-null   float64
 11  Costo Unitario   1010 non-null   float64
 12  Forma de Pago    1010 non-null   object 
 13  Descuento (%)    1010 non-null   int64  
 14  IVA (%)          1004 non-null   float64
dtypes: float64(3), int64(3), object(9)
memory usage: 118.5+ KB


In [14]:
df.describe()

,ID Venta,Cantidad,Precio Unitario,Costo Unitario,Descuento (%),IVA (%)
count,1010.000000,1010.000000,1010.000000,1010.000000,1010.000000,1004.0
mean,499.374257,5.466337,12434.653931,8743.972396,10.990099,16.0
std,289.623789,3.006329,7129.486699,5146.750754,13.542866,0.0
min,1.000000,-10.000000,183.550000,112.600000,0.000000,16.0
25%,247.250000,3.000000,6272.595000,4341.550000,5.000000,16.0
50%,499.000000,6.000000,12297.880000,8513.320000,10.000000,16.0
75%,750.750000,8.000000,18584.012500,12869.877500,15.000000,16.0
max,1000.000000,10.000000,24990.250000,20562.310000,150.000000,16.0


Ejercicio de comprobación
Responde las siguientes preguntas:
1.	¿Cuántos registros contiene el archivo?
1,010 registros.
2.	¿Cuántas columnas tiene?
15 columnas.
3.	¿Cuál es el nombre de la primera columna?
ID Venta.
4.	¿Cuál es el nombre de la última columna?
IVA (%).
5.	¿Qué tipo de dato tiene la columna Cantidad?
int64 (número entero).
6.	¿Qué tipo de dato tiene la columna Precio Unitario?
float64 (número decimal).
7.	¿Qué tipo de dato tiene la columna Fecha?
object (texto).
8.	¿Cuál es el precio unitario máximo?
$24,990.25.
9.	¿Cuál es la cantidad máxima registrada?
10 unidades.
10.	¿Observas algún dato que parezca incorrecto o fuera de lugar?
Sí. Al revisar la estructura del conjunto de datos  pudimos identificar identificar algunos posibles problemas de calidad, por ejemplo:
La columna IVA (%) contiene algunos valores nulos (vacíos).
La columna Fecha está almacenada como texto (object) en lugar de un formato de fecha (datetime), lo que puede dificultar algunos análisis.


Práctica 3

In [15]:
df["Cliente"]

,Cliente
0,Cliente 200
1,Cliente 124
2,Cliente 158
3,Cliente 185
4,Cliente 33
...,...
1005,Cliente 111
1006,Cliente 124
1007,Cliente 89
1008,Cliente 36


In [16]:
df[["Cliente","Producto","Cantidad"]]

,Cliente,Producto,Cantidad
0,Cliente 200,Laptop Lenovo,10
1,Cliente 124,Laptop Dell,2
2,Cliente 158,Monitor Samsung,2
3,Cliente 185,Laptop HP,3
4,Cliente 33,Laptop Dell,6
...,...,...,...
1005,Cliente 111,Audífonos,8
1006,Cliente 124,Laptop Lenovo,7
1007,Cliente 89,Laptop Dell,9
1008,Cliente 36,Laptop Dell,-1


In [17]:
df.iloc[0]

,0
ID Venta,1
Fecha,2026-02-08 00:00:00
Sucursal,Norte
Ciudad,Querétaro
Estado,Querétaro
Vendedor,Luis Torres
Cliente,Cliente 200
Categoría,Laptop
Producto,Laptop Lenovo
Cantidad,10


In [18]:
df.iloc[1]

,1
ID Venta,2
Fecha,2026-04-16 00:00:00
Sucursal,Poniente
Ciudad,Aguascalientes
Estado,Aguascalientes
Vendedor,Luis Torres
Cliente,Cliente 124
Categoría,Laptop
Producto,Laptop Dell
Cantidad,2


In [19]:
df.iloc[0:5]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0
1,2,2026-04-16 00:00:00,Poniente,Aguascalientes,Aguascalientes,Luis Torres,Cliente 124,Laptop,Laptop Dell,2,1231.08,914.96,Tarjeta,10,16.0
2,3,2026-05-21 00:00:00,Poniente,Celaya,Guanajuato,Ana López,Cliente 158,Monitor,Monitor Samsung,2,15290.91,9106.87,Tarjeta,10,16.0
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop HP,3,10756.13,6950.42,Transferencia,20,16.0
4,5,2026-05-19 00:00:00,Sur,Aguascalientes,Aguascalientes,Ana López,Cliente 33,Laptop,Laptop Dell,6,18571.09,12060.69,Transferencia,15,16.0


In [20]:
df.iloc[0:5,0:4]

,ID Venta,Fecha,Sucursal,Ciudad
0,1,2026-02-08 00:00:00,Norte,Querétaro
1,2,2026-04-16 00:00:00,Poniente,Aguascalientes
2,3,2026-05-21 00:00:00,Poniente,Celaya
3,4,2026-02-08 00:00:00,Sur,Querétaro
4,5,2026-05-19 00:00:00,Sur,Aguascalientes


In [21]:
df[df["Ciudad"]=="Querétaro"]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop HP,3,10756.13,6950.42,Transferencia,20,16.0
7,8,2026-02-07 00:00:00,Sur,Querétaro,Querétaro,Luis Torres,Cliente 98,Monitor,Monitor LG,7,14680.86,12243.79,Tarjeta,20,16.0
15,16,2026-06-25 00:00:00,Oriente,Querétaro,Querétaro,María Gómez,Cliente 169,Accesorios,Mouse,3,15577.98,10968.73,Tarjeta,5,16.0
19,20,2026-06-05 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 112,Monitor,Monitor Samsung,1,2673.79,1740.96,Efectivo,10,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
975,976,2026-06-27 00:00:00,Centro,Querétaro,Querétaro,Ana López,Cliente 197,Accesorios,Webcam,10,3028.28,2380.23,Transferencia,0,16.0
991,992,2026-03-29 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 137,Monitor,Monitor Samsung,6,11432.66,9119.33,Transferencia,0,16.0
992,993,2026-02-09 00:00:00,Sur,Querétaro,Querétaro,María Gómez,Cliente 36,Laptop,Laptop HP,9,18893.96,12766.98,Transferencia,20,16.0
1002,862,2026-02-18 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 52,Accesorios,Teclado,2,11606.59,7457.87,Tarjeta,20,16.0


In [22]:
df[df["Categoría"]=="Laptop"]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0
1,2,2026-04-16 00:00:00,Poniente,Aguascalientes,Aguascalientes,Luis Torres,Cliente 124,Laptop,Laptop Dell,2,1231.08,914.96,Tarjeta,10,16.0
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop HP,3,10756.13,6950.42,Transferencia,20,16.0
4,5,2026-05-19 00:00:00,Sur,Aguascalientes,Aguascalientes,Ana López,Cliente 33,Laptop,Laptop Dell,6,18571.09,12060.69,Transferencia,15,16.0
8,9,2026-03-28 00:00:00,Sur,San Luis Potosí,San Luis Potosí,Ana López,Cliente 69,Laptop,Laptop HP,9,18736.06,15318.53,Efectivo,0,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
998,999,2026-05-06 00:00:00,Oriente,León,Guanajuato,María Gómez,Cliente 127,Laptop,Laptop HP,8,6923.12,5342.09,Efectivo,10,16.0
1003,184,2026-03-11 00:00:00,Poniente,San Luis Potosí,San Luis Potosí,Juan Pérez,Cliente 5,Laptop,Laptop Dell,7,12183.91,8112.48,Tarjeta,15,16.0
1006,882,2026-03-16 00:00:00,Centro,Celaya,Guanajuato,Juan Pérez,Cliente 124,Laptop,Laptop Lenovo,7,10036.48,7883.42,Efectivo,0,16.0
1007,499,2026-01-01 00:00:00,Centro,Aguascalientes,Aguascalientes,María Gómez,Cliente 89,Laptop,Laptop Dell,9,18873.15,14705.85,Transferencia,0,16.0


In [23]:
df[df["Precio Unitario"]>10000]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
2,3,2026-05-21 00:00:00,Poniente,Celaya,Guanajuato,Ana López,Cliente 158,Monitor,Monitor Samsung,2,15290.91,9106.87,Tarjeta,10,16.0
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop HP,3,10756.13,6950.42,Transferencia,20,16.0
4,5,2026-05-19 00:00:00,Sur,Aguascalientes,Aguascalientes,Ana López,Cliente 33,Laptop,Laptop Dell,6,18571.09,12060.69,Transferencia,15,16.0
5,6,2026-03-02 00:00:00,Oriente,Aguascalientes,Aguascalientes,carlos ruiz,Cliente 157,Monitor,Monitor Samsung,4,22944.36,15344.97,Tarjeta,15,16.0
7,8,2026-02-07 00:00:00,Sur,Querétaro,Querétaro,Luis Torres,Cliente 98,Monitor,Monitor LG,7,14680.86,12243.79,Tarjeta,20,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1002,862,2026-02-18 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 52,Accesorios,Teclado,2,11606.59,7457.87,Tarjeta,20,16.0
1003,184,2026-03-11 00:00:00,Poniente,San Luis Potosí,San Luis Potosí,Juan Pérez,Cliente 5,Laptop,Laptop Dell,7,12183.91,8112.48,Tarjeta,15,16.0
1004,113,2026-06-23 00:00:00,Poniente,San Luis Potosí,San Luis Potosí,María Gómez,Cliente 157,Accesorios,Teclado,4,21418.19,14151.42,Tarjeta,10,16.0
1006,882,2026-03-16 00:00:00,Centro,Celaya,Guanajuato,Juan Pérez,Cliente 124,Laptop,Laptop Lenovo,7,10036.48,7883.42,Efectivo,0,16.0


In [24]:
df[(df["Ciudad"]=="Querétaro") & (df["Categoría"]=="Laptop")]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop HP,3,10756.13,6950.42,Transferencia,20,16.0
56,57,2026-03-23 00:00:00,Oriente,Querétaro,Querétaro,Sofía Díaz,Cliente 41,Laptop,Laptop HP,3,3334.42,2536.74,Transferencia,0,16.0
80,81,2026-02-19 00:00:00,Norte,Querétaro,Querétaro,María Gómez,Cliente 181,Laptop,Laptop HP,1,11717.04,8099.39,Transferencia,5,16.0
82,83,2026-02-18 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 196,Laptop,Laptop HP,5,22581.88,18772.00,Tarjeta,15,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
950,951,2026-05-13 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 143,Laptop,Laptop Lenovo,2,10237.44,7165.96,Tarjeta,20,16.0
952,953,2026-06-20 00:00:00,Sur,Querétaro,Querétaro,Juan Pérez,Cliente 139,Laptop,Laptop HP,2,16496.94,10288.05,Tarjeta,20,16.0
969,970,2026-04-07 00:00:00,Poniente,Querétaro,Querétaro,María Gómez,Cliente 80,Laptop,Laptop HP,7,4804.09,4023.79,Tarjeta,10,16.0
992,993,2026-02-09 00:00:00,Sur,Querétaro,Querétaro,María Gómez,Cliente 36,Laptop,Laptop HP,9,18893.96,12766.98,Transferencia,20,16.0


In [25]:
df[(df["Ciudad"]=="Querétaro") | (df["Ciudad"]=="León")]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop HP,3,10756.13,6950.42,Transferencia,20,16.0
7,8,2026-02-07 00:00:00,Sur,Querétaro,Querétaro,Luis Torres,Cliente 98,Monitor,Monitor LG,7,14680.86,12243.79,Tarjeta,20,16.0
13,14,2026-01-15 00:00:00,Centro,León,Guanajuato,Sofía Díaz,Cliente 45,Laptop,Laptop HP,5,239.19,181.33,Tarjeta,20,16.0
14,15,2026-05-22 00:00:00,Sur,León,Guanajuato,Carlos Ruiz,Cliente 137,Laptop,Laptop Lenovo,3,9398.82,7924.81,Tarjeta,5,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
998,999,2026-05-06 00:00:00,Oriente,León,Guanajuato,María Gómez,Cliente 127,Laptop,Laptop HP,8,6923.12,5342.09,Efectivo,10,16.0
1001,73,2026-06-19 00:00:00,Oriente,León,Guanajuato,Ana López,Cliente 132,Accesorios,Audífonos,7,8004.53,5707.22,Tarjeta,20,16.0
1002,862,2026-02-18 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 52,Accesorios,Teclado,2,11606.59,7457.87,Tarjeta,20,16.0
1005,898,2026-05-26 00:00:00,Centro,León,Guanajuato,sofía díaz,Cliente 111,Accesorios,Audífonos,8,1076.11,663.89,Efectivo,0,16.0


Ejercicio de comprobación
Responde las siguientes preguntas:
1.	Mostrar todas las ventas realizadas en León.



In [26]:
df[df["Ciudad"]=="León"]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
13,14,2026-01-15 00:00:00,Centro,León,Guanajuato,Sofía Díaz,Cliente 45,Laptop,Laptop HP,5,239.19,181.33,Tarjeta,20,16.0
14,15,2026-05-22 00:00:00,Sur,León,Guanajuato,Carlos Ruiz,Cliente 137,Laptop,Laptop Lenovo,3,9398.82,7924.81,Tarjeta,5,16.0
17,18,2026-06-11 00:00:00,Centro,León,Guanajuato,Carlos Ruiz,Cliente 83,Monitor,Monitor LG,2,10609.24,7472.13,Transferencia,120,16.0
27,28,2026-05-11 00:00:00,Norte,León,Guanajuato,María Gómez,Cliente 54,Laptop,Laptop Dell,9,19227.65,15901.44,Tarjeta,5,16.0
31,32,2026-02-20 00:00:00,Poniente,León,Guanajuato,Ana López,Cliente 137,Monitor,Monitor Samsung,5,14469.72,11188.03,Efectivo,20,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
994,995,2026-02-22 00:00:00,Sur,León,Guanajuato,Carlos Ruiz,Cliente 51,Monitor,Monitor LG,7,16713.86,10135.72,Efectivo,20,16.0
996,997,2026-06-18 00:00:00,Norte,León,Guanajuato,Carlos Ruiz,Cliente 14,Laptop,Laptop Dell,1,16359.32,10535.98,Tarjeta,20,16.0
998,999,2026-05-06 00:00:00,Oriente,León,Guanajuato,María Gómez,Cliente 127,Laptop,Laptop HP,8,6923.12,5342.09,Efectivo,10,16.0
1001,73,2026-06-19 00:00:00,Oriente,León,Guanajuato,Ana López,Cliente 132,Accesorios,Audífonos,7,8004.53,5707.22,Tarjeta,20,16.0


2.	Mostrar únicamente la columna Vendedor.

In [27]:
df["Vendedor"]

,Vendedor
0,Luis Torres
1,Luis Torres
2,Ana López
3,Sofía Díaz
4,Ana López
...,...
1005,sofía díaz
1006,Juan Pérez
1007,María Gómez
1008,Sofía Díaz


3.	Mostrar las columnas Cliente, Producto y Cantidad.

In [28]:
df[["Cliente","Producto","Cantidad"]]

,Cliente,Producto,Cantidad
0,Cliente 200,Laptop Lenovo,10
1,Cliente 124,Laptop Dell,2
2,Cliente 158,Monitor Samsung,2
3,Cliente 185,Laptop HP,3
4,Cliente 33,Laptop Dell,6
...,...,...,...
1005,Cliente 111,Audífonos,8
1006,Cliente 124,Laptop Lenovo,7
1007,Cliente 89,Laptop Dell,9
1008,Cliente 36,Laptop Dell,-1


4. Mostrar las ventas con precio unitario mayor a $20,000.

In [29]:
df[df["Precio Unitario"]>20000]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
5,6,2026-03-02 00:00:00,Oriente,Aguascalientes,Aguascalientes,carlos ruiz,Cliente 157,Monitor,Monitor Samsung,4,22944.36,15344.97,Tarjeta,15,16.0
11,12,2026-01-17 00:00:00,Sur,San Luis Potosí,San Luis Potosí,Carlos Ruiz,Cliente 52,Accesorios,Webcam,4,24257.31,15304.98,Transferencia,0,16.0
20,21,2026-01-20 00:00:00,Poniente,Querétaro,Querétaro,Carlos Ruiz,Cliente 188,Monitor,Monitor Samsung,5,21685.93,14372.42,Efectivo,5,16.0
21,22,2026-06-11 00:00:00,Centro,Celaya,Guanajuato,María Gómez,Cliente 68,Laptop,Laptop Dell,1,21317.39,13092.21,Tarjeta,0,16.0
24,25,2026-05-20 00:00:00,Oriente,Aguascalientes,Aguascalientes,Ana López,Cliente 143,Laptop,Laptop Dell,9,21995.43,17966.34,Tarjeta,0,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
967,968,2026-03-29 00:00:00,Sur,León,Guanajuato,Luis Torres,Cliente 92,Monitor,Monitor Samsung,4,24154.95,16357.12,Tarjeta,0,16.0
970,971,14/01/2026,Poniente,Celaya,Guanajuato,Juan Pérez,Cliente 148,Monitor,Monitor LG,7,21157.68,13942.83,Efectivo,5,16.0
989,990,2026-01-06 00:00:00,Sur,León,Guanajuato,Carlos Ruiz,Cliente 178,Laptop,Laptop Lenovo,5,20358.72,14817.06,Efectivo,20,16.0
990,991,2026-05-13 00:00:00,Oriente,Aguascalientes,Aguascalientes,Sofía Díaz,Cliente 38,Monitor,Monitor Samsung,5,23097.80,16427.60,Tarjeta,5,16.0


5. Mostrar las ventas cuya categoría sea Monitor.

In [30]:
df[df["Categoría"]=="Monitor"]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
2,3,2026-05-21 00:00:00,Poniente,Celaya,Guanajuato,Ana López,Cliente 158,Monitor,Monitor Samsung,2,15290.91,9106.87,Tarjeta,10,16.0
5,6,2026-03-02 00:00:00,Oriente,Aguascalientes,Aguascalientes,carlos ruiz,Cliente 157,Monitor,Monitor Samsung,4,22944.36,15344.97,Tarjeta,15,16.0
6,7,2026-02-02 00:00:00,Oriente,Aguascalientes,Aguascalientes,Juan Pérez,Cliente 189,Monitor,Monitor LG,10,206.32,173.02,Tarjeta,10,16.0
7,8,2026-02-07 00:00:00,Sur,Querétaro,Querétaro,Luis Torres,Cliente 98,Monitor,Monitor LG,7,14680.86,12243.79,Tarjeta,20,16.0
9,10,2026-01-17 00:00:00,Poniente,Celaya,Guanajuato,Sofía Díaz,Cliente 27,Monitor,Monitor Samsung,9,15887.35,11778.45,Tarjeta,15,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
991,992,2026-03-29 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 137,Monitor,Monitor Samsung,6,11432.66,9119.33,Transferencia,0,16.0
993,994,2026-06-16 00:00:00,Centro,León,Guanajuato,Juan Pérez,Cliente 162,Monitor,Monitor LG,9,15192.21,12063.38,Efectivo,10,16.0
994,995,2026-02-22 00:00:00,Sur,León,Guanajuato,Carlos Ruiz,Cliente 51,Monitor,Monitor LG,7,16713.86,10135.72,Efectivo,20,16.0
995,996,2026-06-04 00:00:00,Norte,Celaya,Guanajuato,Ana López,Cliente 105,Monitor,Monitor Samsung,8,1920.24,1146.18,Tarjeta,20,16.0


6. Mostrar las ventas realizadas en Querétaro de la categoría Laptop.

In [31]:
df[(df["Ciudad"]=="Querétaro") & (df["Categoría"]=="Laptop")]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop HP,3,10756.13,6950.42,Transferencia,20,16.0
56,57,2026-03-23 00:00:00,Oriente,Querétaro,Querétaro,Sofía Díaz,Cliente 41,Laptop,Laptop HP,3,3334.42,2536.74,Transferencia,0,16.0
80,81,2026-02-19 00:00:00,Norte,Querétaro,Querétaro,María Gómez,Cliente 181,Laptop,Laptop HP,1,11717.04,8099.39,Transferencia,5,16.0
82,83,2026-02-18 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 196,Laptop,Laptop HP,5,22581.88,18772.00,Tarjeta,15,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
950,951,2026-05-13 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 143,Laptop,Laptop Lenovo,2,10237.44,7165.96,Tarjeta,20,16.0
952,953,2026-06-20 00:00:00,Sur,Querétaro,Querétaro,Juan Pérez,Cliente 139,Laptop,Laptop HP,2,16496.94,10288.05,Tarjeta,20,16.0
969,970,2026-04-07 00:00:00,Poniente,Querétaro,Querétaro,María Gómez,Cliente 80,Laptop,Laptop HP,7,4804.09,4023.79,Tarjeta,10,16.0
992,993,2026-02-09 00:00:00,Sur,Querétaro,Querétaro,María Gómez,Cliente 36,Laptop,Laptop HP,9,18893.96,12766.98,Transferencia,20,16.0


7. Mostrar las ventas cuyo descuento sea mayor al 10%.

In [32]:
df[df["Descuento (%)"]>10]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop HP,3,10756.13,6950.42,Transferencia,20,16.0
4,5,2026-05-19 00:00:00,Sur,Aguascalientes,Aguascalientes,Ana López,Cliente 33,Laptop,Laptop Dell,6,18571.09,12060.69,Transferencia,15,16.0
5,6,2026-03-02 00:00:00,Oriente,Aguascalientes,Aguascalientes,carlos ruiz,Cliente 157,Monitor,Monitor Samsung,4,22944.36,15344.97,Tarjeta,15,16.0
7,8,2026-02-07 00:00:00,Sur,Querétaro,Querétaro,Luis Torres,Cliente 98,Monitor,Monitor LG,7,14680.86,12243.79,Tarjeta,20,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
999,1000,2026-06-15 00:00:00,Centro,Celaya,Guanajuato,Carlos Ruiz,Cliente 198,Monitor,Monitor Samsung,1,13486.85,8901.45,Efectivo,20,16.0
1001,73,2026-06-19 00:00:00,Oriente,León,Guanajuato,Ana López,Cliente 132,Accesorios,Audífonos,7,8004.53,5707.22,Tarjeta,20,16.0
1002,862,2026-02-18 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 52,Accesorios,Teclado,2,11606.59,7457.87,Tarjeta,20,16.0
1003,184,2026-03-11 00:00:00,Poniente,San Luis Potosí,San Luis Potosí,Juan Pérez,Cliente 5,Laptop,Laptop Dell,7,12183.91,8112.48,Tarjeta,15,16.0


8. Mostrar las ventas pagadas con Tarjeta.

In [33]:
df[df["Forma de Pago"]=="Tarjeta"]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0
1,2,2026-04-16 00:00:00,Poniente,Aguascalientes,Aguascalientes,Luis Torres,Cliente 124,Laptop,Laptop Dell,2,1231.08,914.96,Tarjeta,10,16.0
2,3,2026-05-21 00:00:00,Poniente,Celaya,Guanajuato,Ana López,Cliente 158,Monitor,Monitor Samsung,2,15290.91,9106.87,Tarjeta,10,16.0
5,6,2026-03-02 00:00:00,Oriente,Aguascalientes,Aguascalientes,carlos ruiz,Cliente 157,Monitor,Monitor Samsung,4,22944.36,15344.97,Tarjeta,15,16.0
6,7,2026-02-02 00:00:00,Oriente,Aguascalientes,Aguascalientes,Juan Pérez,Cliente 189,Monitor,Monitor LG,10,206.32,173.02,Tarjeta,10,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1001,73,2026-06-19 00:00:00,Oriente,León,Guanajuato,Ana López,Cliente 132,Accesorios,Audífonos,7,8004.53,5707.22,Tarjeta,20,16.0
1002,862,2026-02-18 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 52,Accesorios,Teclado,2,11606.59,7457.87,Tarjeta,20,16.0
1003,184,2026-03-11 00:00:00,Poniente,San Luis Potosí,San Luis Potosí,Juan Pérez,Cliente 5,Laptop,Laptop Dell,7,12183.91,8112.48,Tarjeta,15,16.0
1004,113,2026-06-23 00:00:00,Poniente,San Luis Potosí,San Luis Potosí,María Gómez,Cliente 157,Accesorios,Teclado,4,21418.19,14151.42,Tarjeta,10,16.0


9. Mostrar las ventas cuya cantidad sea menor a 3.

In [34]:
df[df["Cantidad"]<3]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
1,2,2026-04-16 00:00:00,Poniente,Aguascalientes,Aguascalientes,Luis Torres,Cliente 124,Laptop,Laptop Dell,2,1231.08,914.96,Tarjeta,10,16.0
2,3,2026-05-21 00:00:00,Poniente,Celaya,Guanajuato,Ana López,Cliente 158,Monitor,Monitor Samsung,2,15290.91,9106.87,Tarjeta,10,16.0
17,18,2026-06-11 00:00:00,Centro,León,Guanajuato,Carlos Ruiz,Cliente 83,Monitor,Monitor LG,2,10609.24,7472.13,Transferencia,120,16.0
19,20,2026-06-05 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 112,Monitor,Monitor Samsung,1,2673.79,1740.96,Efectivo,10,16.0
21,22,2026-06-11 00:00:00,Centro,Celaya,Guanajuato,María Gómez,Cliente 68,Laptop,Laptop Dell,1,21317.39,13092.21,Tarjeta,0,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
996,997,2026-06-18 00:00:00,Norte,León,Guanajuato,Carlos Ruiz,Cliente 14,Laptop,Laptop Dell,1,16359.32,10535.98,Tarjeta,20,16.0
997,998,2026-03-17 00:00:00,Poniente,San Luis Potosí,San Luis Potosí,Sofía Díaz,Cliente 159,Accesorios,Mouse,1,10263.69,8086.38,Tarjeta,15,16.0
999,1000,2026-06-15 00:00:00,Centro,Celaya,Guanajuato,Carlos Ruiz,Cliente 198,Monitor,Monitor Samsung,1,13486.85,8901.45,Efectivo,20,16.0
1002,862,2026-02-18 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 52,Accesorios,Teclado,2,11606.59,7457.87,Tarjeta,20,16.0


10. Mostrar las ventas de la sucursal Centro.

In [35]:
df[df["Sucursal"]=="Centro"]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
13,14,2026-01-15 00:00:00,Centro,León,Guanajuato,Sofía Díaz,Cliente 45,Laptop,Laptop HP,5,239.19,181.33,Tarjeta,20,16.0
17,18,2026-06-11 00:00:00,Centro,León,Guanajuato,Carlos Ruiz,Cliente 83,Monitor,Monitor LG,2,10609.24,7472.13,Transferencia,120,16.0
19,20,2026-06-05 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 112,Monitor,Monitor Samsung,1,2673.79,1740.96,Efectivo,10,16.0
21,22,2026-06-11 00:00:00,Centro,Celaya,Guanajuato,María Gómez,Cliente 68,Laptop,Laptop Dell,1,21317.39,13092.21,Tarjeta,0,16.0
25,26,2026-02-14 00:00:00,Centro,Celaya,Guanajuato,Ana López,Cliente 39,Accesorios,Webcam,5,17986.43,12415.31,Efectivo,20,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
999,1000,2026-06-15 00:00:00,Centro,Celaya,Guanajuato,Carlos Ruiz,Cliente 198,Monitor,Monitor Samsung,1,13486.85,8901.45,Efectivo,20,16.0
1002,862,2026-02-18 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 52,Accesorios,Teclado,2,11606.59,7457.87,Tarjeta,20,16.0
1005,898,2026-05-26 00:00:00,Centro,León,Guanajuato,sofía díaz,Cliente 111,Accesorios,Audífonos,8,1076.11,663.89,Efectivo,0,16.0
1006,882,2026-03-16 00:00:00,Centro,Celaya,Guanajuato,Juan Pérez,Cliente 124,Laptop,Laptop Lenovo,7,10036.48,7883.42,Efectivo,0,16.0


PRACTICA 4

In [36]:
df.isnull()

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
1,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
3,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
4,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1005,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
1006,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
1007,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
1008,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False


In [37]:
df.isnull().sum()

,0
ID Venta,0
Fecha,0
Sucursal,0
Ciudad,0
Estado,0
Vendedor,0
Cliente,0
Categoría,0
Producto,0
Cantidad,0


In [38]:
df.duplicated()

,0
0,False
1,False
2,False
3,False
4,False
...,...
1005,True
1006,True
1007,True
1008,True


In [39]:
df.duplicated().sum()

np.int64(10)

In [40]:
df[df.duplicated()]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
1000,128,2026-04-01 00:00:00,Sur,Aguascalientes,Aguascalientes,Sofía Díaz,Cliente 179,Accesorios,Teclado,7,11463.14,7479.55,Efectivo,5,16.0
1001,73,2026-06-19 00:00:00,Oriente,León,Guanajuato,Ana López,Cliente 132,Accesorios,Audífonos,7,8004.53,5707.22,Tarjeta,20,16.0
1002,862,2026-02-18 00:00:00,Centro,Querétaro,Querétaro,María Gómez,Cliente 52,Accesorios,Teclado,2,11606.59,7457.87,Tarjeta,20,16.0
1003,184,2026-03-11 00:00:00,Poniente,San Luis Potosí,San Luis Potosí,Juan Pérez,Cliente 5,Laptop,Laptop Dell,7,12183.91,8112.48,Tarjeta,15,16.0
1004,113,2026-06-23 00:00:00,Poniente,San Luis Potosí,San Luis Potosí,María Gómez,Cliente 157,Accesorios,Teclado,4,21418.19,14151.42,Tarjeta,10,16.0
1005,898,2026-05-26 00:00:00,Centro,León,Guanajuato,sofía díaz,Cliente 111,Accesorios,Audífonos,8,1076.11,663.89,Efectivo,0,16.0
1006,882,2026-03-16 00:00:00,Centro,Celaya,Guanajuato,Juan Pérez,Cliente 124,Laptop,Laptop Lenovo,7,10036.48,7883.42,Efectivo,0,16.0
1007,499,2026-01-01 00:00:00,Centro,Aguascalientes,Aguascalientes,María Gómez,Cliente 89,Laptop,Laptop Dell,9,18873.15,14705.85,Transferencia,0,16.0
1008,135,2026-03-04 00:00:00,Oriente,Querétaro,Querétaro,Sofía Díaz,Cliente 36,Laptop,Laptop Dell,-1,4668.93,3620.85,Tarjeta,5,16.0
1009,94,2026-03-06 00:00:00,Sur,Aguascalientes,Aguascalientes,Juan Pérez,Cliente 129,Accesorios,Webcam,8,7441.66,4620.54,Transferencia,15,16.0


In [41]:
df["Ciudad"].unique()

array(['Querétaro', 'Aguascalientes', 'Celaya', 'San Luis Potosí', 'León',
       'Queretaro', 'queretaro', 'QUERETARO'], dtype=object)

In [42]:
df["Ciudad"].value_counts()

,count
Ciudad,
Aguascalientes,213
Querétaro,211
Celaya,209
San Luis Potosí,190
León,184
Queretaro,1
queretaro,1
QUERETARO,1


In [43]:
df["Forma de Pago"].value_counts()

,count
Forma de Pago,
Transferencia,347
Tarjeta,337
Efectivo,326


In [44]:
df[df["Cantidad"] < 0]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
134,135,2026-03-04 00:00:00,Oriente,Querétaro,Querétaro,Sofía Díaz,Cliente 36,Laptop,Laptop Dell,-1,4668.93,3620.85,Tarjeta,5,16.0
177,178,2026-03-04 00:00:00,Centro,Aguascalientes,Aguascalientes,Luis Torres,Cliente 134,Accesorios,Teclado,-10,15752.54,13201.09,Tarjeta,10,16.0
182,183,2026-06-06 00:00:00,Oriente,Celaya,Guanajuato,Sofía Díaz,Cliente 51,Accesorios,Webcam,-1,20898.30,13244.08,Transferencia,10,16.0
240,241,2026-06-19 00:00:00,Oriente,Celaya,Guanajuato,Sofía Díaz,Cliente 85,Laptop,Laptop Dell,-2,5135.76,3008.09,Transferencia,0,16.0
248,249,2026-05-28 00:00:00,Sur,San Luis Potosí,San Luis Potosí,Juan Pérez,Cliente 27,Accesorios,Mouse,-6,11284.66,8247.47,Efectivo,15,16.0
374,375,2026-03-31 00:00:00,Poniente,Celaya,Guanajuato,Sofía Díaz,Cliente 132,Laptop,Laptop Dell,-3,14077.60,9382.02,Efectivo,10,16.0
438,439,2026-01-11 00:00:00,Norte,Celaya,Guanajuato,Luis Torres,Cliente 193,Monitor,Monitor LG,-4,3233.86,2719.08,Tarjeta,15,16.0
634,635,2026-03-20 00:00:00,Norte,San Luis Potosí,San Luis Potosí,Luis Torres,Cliente 22,Accesorios,Webcam,-6,21242.71,15346.50,Transferencia,20,16.0
1008,135,2026-03-04 00:00:00,Oriente,Querétaro,Querétaro,Sofía Díaz,Cliente 36,Laptop,Laptop Dell,-1,4668.93,3620.85,Tarjeta,5,16.0


In [45]:
df[df["Descuento (%)"] > 100]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
17,18,2026-06-11 00:00:00,Centro,León,Guanajuato,Carlos Ruiz,Cliente 83,Monitor,Monitor LG,2,10609.24,7472.13,Transferencia,120,16.0
419,420,2026-05-17 00:00:00,Poniente,León,Guanajuato,María Gómez,Cliente 198,Laptop,Laptop Lenovo,8,15620.28,9441.61,Tarjeta,120,16.0
455,456,2026-03-05 00:00:00,Oriente,León,Guanajuato,Carlos Ruiz,Cliente 12,Monitor,Monitor Samsung,9,2332.31,1538.99,Efectivo,150,16.0
520,521,2026-06-05 00:00:00,Oriente,Aguascalientes,Aguascalientes,Luis Torres,Cliente 55,Laptop,Laptop Dell,1,16065.87,9598.85,Efectivo,150,16.0
529,530,2026-04-21 00:00:00,Sur,León,Guanajuato,María Gómez,Cliente 70,Accesorios,Audífonos,10,15968.27,13443.89,Efectivo,150,16.0
646,647,2026-04-20 00:00:00,Oriente,San Luis Potosí,San Luis Potosí,Luis Torres,Cliente 19,Laptop,Laptop Lenovo,9,15905.75,12017.34,Transferencia,150,16.0
740,741,2026-01-29 00:00:00,Norte,San Luis Potosí,San Luis Potosí,Ana López,Cliente 143,Laptop,Laptop Lenovo,9,1684.95,952.54,Tarjeta,150,16.0
957,958,2026-03-23 00:00:00,Poniente,León,Guanajuato,Juan Pérez,Cliente 102,Accesorios,Teclado,1,3203.41,2514.05,Efectivo,120,16.0


In [46]:
df[df["IVA (%)"].isnull()]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
59,60,2026-02-02 00:00:00,Poniente,León,Guanajuato,Luis Torres,Cliente 172,Accesorios,Teclado,4,22210.39,15095.64,Tarjeta,20,NaN
133,134,2026-06-28 00:00:00,Norte,San Luis Potosí,San Luis Potosí,Luis Torres,Cliente 165,Laptop,Laptop Lenovo,8,18194.90,13090.08,Tarjeta,10,NaN
378,379,2026-04-17 00:00:00,Norte,Aguascalientes,Aguascalientes,María Gómez,Cliente 114,Monitor,Monitor LG,4,1537.83,1027.84,Transferencia,5,NaN
392,393,2026-04-03 00:00:00,Oriente,San Luis Potosí,San Luis Potosí,Juan Pérez,Cliente 156,Accesorios,Mouse,2,17041.11,12967.60,Efectivo,0,NaN
633,634,2026-02-17 00:00:00,Poniente,Aguascalientes,Aguascalientes,Sofía Díaz,Cliente 36,Laptop,Laptop Lenovo,10,19137.49,13210.21,Tarjeta,5,NaN
962,963,2026-06-11 00:00:00,Poniente,León,Guanajuato,María Gómez,Cliente 37,Monitor,Monitor LG,10,11495.17,9074.76,Transferencia,5,NaN


1.	¿Cuántos registros duplicados tiene el archivo?
10
2.	¿Qué columnas contienen valores nulos?
IVA%
3.	¿Cuántos registros tienen el IVA vacío?
6
4.	¿Cuántas ciudades diferentes aparecen registradas?
8
5.	¿Qué ciudad tiene más ventas?
Aguascalientes
6.	¿Cuántas ventas tienen una cantidad negativa?
9 ventas
7.	¿Cuántas ventas presentan descuentos mayores al 100%?
8
8.	¿Qué formas de pago existen en el archivo?
tarjeta, efectivo, transferencia
9.	¿Cuál es la categoría con más registros?
laptop
10.	¿Qué problemas de calidad detectaste además de los anteriores?
Existen diferentes formas de escribir Querétaro (Querétaro, Queretaro, queretaro y QUERETARO), lo que genera inconsistencias.
Hay registros con cantidades negativas, que probablemente son errores de captura.
Existen registros con descuentos mayores al 100%, los cuales no son válidos para una venta.


**PRACTICA 5**

In [48]:
df_limpio = df.copy()


In [49]:
df_limpio.shape

(1010, 15)

In [50]:
df_limpio = df_limpio.drop_duplicates()

In [51]:
df_limpio.shape

(1000, 15)

¿Disminuyó el número de registros? Disminuyó

In [52]:
df_limpio["Cliente"] = df_limpio["Cliente"].str.strip()

In [53]:
df_limpio["Vendedor"] = df_limpio["Vendedor"].str.title()

In [54]:
df_limpio["Ciudad"] = df_limpio["Ciudad"].replace({
    "QUERETARO":"Querétaro",
    "Queretaro":"Querétaro",
    "queretaro":"Querétaro",
    "Qro":"Querétaro"
})


¿Qué ocurrió con las variantes?
Todas las variantes (Queretaro, queretaro, QUERETARO y Qro) fueron reemplazadas por un único nombre: Querétaro, dejando un solo valor consistente para la ciudad.

In [55]:
df_limpio["Producto"] = df_limpio["Producto"].str.title()

In [56]:
df_limpio[df_limpio["Cantidad"] < 0]

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
134,135,2026-03-04 00:00:00,Oriente,Querétaro,Querétaro,Sofía Díaz,Cliente 36,Laptop,Laptop Dell,-1,4668.93,3620.85,Tarjeta,5,16.0
177,178,2026-03-04 00:00:00,Centro,Aguascalientes,Aguascalientes,Luis Torres,Cliente 134,Accesorios,Teclado,-10,15752.54,13201.09,Tarjeta,10,16.0
182,183,2026-06-06 00:00:00,Oriente,Celaya,Guanajuato,Sofía Díaz,Cliente 51,Accesorios,Webcam,-1,20898.30,13244.08,Transferencia,10,16.0
240,241,2026-06-19 00:00:00,Oriente,Celaya,Guanajuato,Sofía Díaz,Cliente 85,Laptop,Laptop Dell,-2,5135.76,3008.09,Transferencia,0,16.0
248,249,2026-05-28 00:00:00,Sur,San Luis Potosí,San Luis Potosí,Juan Pérez,Cliente 27,Accesorios,Mouse,-6,11284.66,8247.47,Efectivo,15,16.0
374,375,2026-03-31 00:00:00,Poniente,Celaya,Guanajuato,Sofía Díaz,Cliente 132,Laptop,Laptop Dell,-3,14077.60,9382.02,Efectivo,10,16.0
438,439,2026-01-11 00:00:00,Norte,Celaya,Guanajuato,Luis Torres,Cliente 193,Monitor,Monitor Lg,-4,3233.86,2719.08,Tarjeta,15,16.0
634,635,2026-03-20 00:00:00,Norte,San Luis Potosí,San Luis Potosí,Luis Torres,Cliente 22,Accesorios,Webcam,-6,21242.71,15346.50,Transferencia,20,16.0


In [57]:
df_limpio = df_limpio[df_limpio["Cantidad"] >= 0]

¿Por qué decidimos eliminarlos y no convertirlos en positivos?
Porque una cantidad negativa representa un error de captura. Convertirla a positiva modificaría la información original y podría generar resultados incorrectos. Por ello, es mejor eliminar esos registros, de acuerdo con las reglas del negocio.

In [58]:
df_limpio = df_limpio[df_limpio["Descuento (%)"] <= 100]

In [59]:
df_limpio["IVA (%)"] = df_limpio["IVA (%)"].fillna(16)

In [60]:
df_limpio.isnull().sum()

,0
ID Venta,0
Fecha,0
Sucursal,0
Ciudad,0
Estado,0
Vendedor,0
Cliente,0
Categoría,0
Producto,0
Cantidad,0


In [61]:
df_limpio.duplicated().sum()

np.int64(0)

In [62]:
df_limpio["Ciudad"].value_counts()

,count
Ciudad,
Querétaro,211
Aguascalientes,208
Celaya,204
San Luis Potosí,184
León,177


PRACTICA **6**

In [63]:
df_limpio

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%)
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0
1,2,2026-04-16 00:00:00,Poniente,Aguascalientes,Aguascalientes,Luis Torres,Cliente 124,Laptop,Laptop Dell,2,1231.08,914.96,Tarjeta,10,16.0
2,3,2026-05-21 00:00:00,Poniente,Celaya,Guanajuato,Ana López,Cliente 158,Monitor,Monitor Samsung,2,15290.91,9106.87,Tarjeta,10,16.0
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop Hp,3,10756.13,6950.42,Transferencia,20,16.0
4,5,2026-05-19 00:00:00,Sur,Aguascalientes,Aguascalientes,Ana López,Cliente 33,Laptop,Laptop Dell,6,18571.09,12060.69,Transferencia,15,16.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
995,996,2026-06-04 00:00:00,Norte,Celaya,Guanajuato,Ana López,Cliente 105,Monitor,Monitor Samsung,8,1920.24,1146.18,Tarjeta,20,16.0
996,997,2026-06-18 00:00:00,Norte,León,Guanajuato,Carlos Ruiz,Cliente 14,Laptop,Laptop Dell,1,16359.32,10535.98,Tarjeta,20,16.0
997,998,2026-03-17 00:00:00,Poniente,San Luis Potosí,San Luis Potosí,Sofía Díaz,Cliente 159,Accesorios,Mouse,1,10263.69,8086.38,Tarjeta,15,16.0
998,999,2026-05-06 00:00:00,Oriente,León,Guanajuato,María Gómez,Cliente 127,Laptop,Laptop Hp,8,6923.12,5342.09,Efectivo,10,16.0


In [64]:
df_limpio["Subtotal"] = df_limpio["Cantidad"] * df_limpio["Precio Unitario"]

In [65]:
df_limpio.head()

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,Precio Unitario,Costo Unitario,Forma de Pago,Descuento (%),IVA (%),Subtotal
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,9461.69,6066.88,Tarjeta,15,16.0,94616.90
1,2,2026-04-16 00:00:00,Poniente,Aguascalientes,Aguascalientes,Luis Torres,Cliente 124,Laptop,Laptop Dell,2,1231.08,914.96,Tarjeta,10,16.0,2462.16
2,3,2026-05-21 00:00:00,Poniente,Celaya,Guanajuato,Ana López,Cliente 158,Monitor,Monitor Samsung,2,15290.91,9106.87,Tarjeta,10,16.0,30581.82
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop Hp,3,10756.13,6950.42,Transferencia,20,16.0,32268.39
4,5,2026-05-19 00:00:00,Sur,Aguascalientes,Aguascalientes,Ana López,Cliente 33,Laptop,Laptop Dell,6,18571.09,12060.69,Transferencia,15,16.0,111426.54


In [66]:
df_limpio["Monto Descuento"] = (
    df_limpio["Subtotal"] *
    df_limpio["Descuento (%)"]/100
)


In [67]:
df_limpio["Subtotal Neto"] = (
    df_limpio["Subtotal"] -
    df_limpio["Monto Descuento"]
)


In [68]:
df_limpio["IVA"] = (
    df_limpio["Subtotal Neto"] *
    df_limpio["IVA (%)"]/100
)


In [69]:
df_limpio["Total"] = (
    df_limpio["Subtotal Neto"] +
    df_limpio["IVA"]
)


In [71]:
df_limpio["Costo Total"] = (
    df_limpio["Cantidad"] *
    df_limpio["Costo Unitario"]
)


In [72]:
df_limpio["Utilidad"] = (
    df_limpio["Subtotal Neto"] -
    df_limpio["Costo Total"]
)


In [73]:
df_limpio[
[
"Cantidad",
"Precio Unitario",
"Subtotal",
"Monto Descuento",
"Subtotal Neto",
"IVA",
"Total",
"Costo Total",
"Utilidad"
]
].head()


,Cantidad,Precio Unitario,Subtotal,Monto Descuento,Subtotal Neto,IVA,Total,Costo Total,Utilidad
0,10,9461.69,94616.90,14192.535,80424.365,12867.89840,93292.26340,60668.80,19755.565
1,2,1231.08,2462.16,246.216,2215.944,354.55104,2570.49504,1829.92,386.024
2,2,15290.91,30581.82,3058.182,27523.638,4403.78208,31927.42008,18213.74,9309.898
3,3,10756.13,32268.39,6453.678,25814.712,4130.35392,29945.06592,20851.26,4963.452
4,6,18571.09,111426.54,16713.981,94712.559,15154.00944,109866.56844,72364.14,22348.419


In [74]:
df_limpio = df_limpio.round(2)

In [75]:
df_limpio.head()

,ID Venta,Fecha,Sucursal,Ciudad,Estado,Vendedor,Cliente,Categoría,Producto,Cantidad,...,Forma de Pago,Descuento (%),IVA (%),Subtotal,Monto Descuento,Subtotal Neto,IVA,Total,Costo Total,Utilidad
0,1,2026-02-08 00:00:00,Norte,Querétaro,Querétaro,Luis Torres,Cliente 200,Laptop,Laptop Lenovo,10,...,Tarjeta,15,16.0,94616.90,14192.54,80424.37,12867.90,93292.26,60668.80,19755.57
1,2,2026-04-16 00:00:00,Poniente,Aguascalientes,Aguascalientes,Luis Torres,Cliente 124,Laptop,Laptop Dell,2,...,Tarjeta,10,16.0,2462.16,246.22,2215.94,354.55,2570.50,1829.92,386.02
2,3,2026-05-21 00:00:00,Poniente,Celaya,Guanajuato,Ana López,Cliente 158,Monitor,Monitor Samsung,2,...,Tarjeta,10,16.0,30581.82,3058.18,27523.64,4403.78,31927.42,18213.74,9309.90
3,4,2026-02-08 00:00:00,Sur,Querétaro,Querétaro,Sofía Díaz,Cliente 185,Laptop,Laptop Hp,3,...,Transferencia,20,16.0,32268.39,6453.68,25814.71,4130.35,29945.07,20851.26,4963.45
4,5,2026-05-19 00:00:00,Sur,Aguascalientes,Aguascalientes,Ana López,Cliente 33,Laptop,Laptop Dell,6,...,Transferencia,15,16.0,111426.54,16713.98,94712.56,15154.01,109866.57,72364.14,22348.42


PRACTICA **7**

In [76]:
df_limpio["Total"].sum()

np.float64(70026165.38)

In [77]:
df_limpio["Total"].mean()

np.float64(71164.80221544715)

In [78]:
df_limpio["Total"].max()

278366.59

In [79]:
df_limpio["Total"].min()

340.67

In [80]:
df_limpio["ID Venta"].count()

np.int64(984)

In [81]:
df_limpio["Cliente"].nunique()

199

In [82]:
df_limpio["Producto"].nunique()

9

In [83]:
df_limpio["Vendedor"].nunique()

6

In [84]:
df_limpio["Sucursal"].nunique()

5

Actividad

1.	¿Cuál fue el total vendido?
2.	¿Cuál fue el promedio por venta?
3.	¿Cuál fue la venta más alta?
4.	¿Cuál fue la venta más baja?
5.	¿Cuántos clientes diferentes compraron?
6.	¿Cuántos productos se vendieron?
7.	¿Cuántos vendedores existen?
8.	¿Cuántas sucursales aparecen en el archivo?


Práctica **8**

In [85]:
df_limpio.groupby("Sucursal")["Total"].sum()

,Total
Sucursal,
Centro,11724331.85
Norte,15270297.94
Oriente,12646219.28
Poniente,15925020.11
Sur,14460296.20


In [86]:
df_limpio.groupby("Ciudad")["Total"].sum()

,Total
Ciudad,
Aguascalientes,15222892.40
Celaya,13917232.11
León,13597313.76
Querétaro,14311193.50
San Luis Potosí,12977533.61


¿Qué ciudad generó mayores ingresos? Aguascalientes

In [87]:
df_limpio.groupby("Vendedor")["Total"].sum()

,Total
Vendedor,
Ana López,10779474.72
Carlos Ruiz,11119336.08
Juan Pérez,12674498.86
Luis Torres,11545260.56
María Gómez,12356718.85
Sofía Díaz,11550876.31


In [88]:
df_limpio.groupby("Vendedor")["Utilidad"].sum()

,Utilidad
Vendedor,
Ana López,1964535.62
Carlos Ruiz,2085655.88
Juan Pérez,2310128.70
Luis Torres,2136338.64
María Gómez,2411064.21
Sofía Díaz,2181648.57


El vendedor con mayores ventas también obtuvo la mayor utilidad?
si

In [89]:
df_limpio.groupby("Categoría")["Total"].sum()

,Total
Categoría,
Accesorios,22961205.80
Laptop,25351003.46
Monitor,21713956.12


In [90]:
df_limpio.groupby("Forma de Pago")["ID Venta"].count()

,ID Venta
Forma de Pago,
Efectivo,317
Tarjeta,327
Transferencia,340


In [91]:
df_limpio.groupby("Sucursal")["Total"].mean()

,Total
Sucursal,
Centro,65135.176944
Norte,78712.875979
Oriente,70256.773778
Poniente,68642.328060
Sur,73031.798990


In [92]:
df_limpio.groupby("Sucursal")["Total"].sum().sort_values(ascending=False)

,Total
Sucursal,
Poniente,15925020.11
Norte,15270297.94
Sur,14460296.20
Oriente,12646219.28
Centro,11724331.85


In [93]:
df_limpio.groupby("Sucursal")["Total"].agg(["sum","mean","max","min"])

,sum,mean,max,min
Sucursal,,,,
Centro,11724331.85,65135.176944,278366.59,1109.84
Norte,15270297.94,78712.875979,272221.96,340.67
Oriente,12646219.28,70256.773778,237899.43,635.51
Poniente,15925020.11,68642.328060,269026.96,1049.77
Sur,14460296.20,73031.798990,267560.50,692.38


Actividad:
Responder:
1.	¿Qué sucursal vendió más?
$15,925,020.11.
2.	¿Qué ciudad obtuvo mayores ingresos?
Aguascalientes, con $15,222,892.40
3.	¿Qué vendedor generó más ventas?
Juan Pérez, con $12,674,498.86 en ventas
4.	¿Qué vendedor obtuvo mayor utilidad?
María Gómez, con una utilidad de $2,411,064.21.
5.	¿Cuál fue la categoría con mayores ingresos?
Laptop, con $25,351,003.46.
6.	¿Cuál es la forma de pago más utilizada?
transferencia
7.	¿Cuál es el promedio de venta por sucursal?

